spikeinterface/sortingcomponents/peak_localization.py

```diff
@@ -162,28 +168,34 @@ def _localize_peaks_chunk(segment_index, start_frame, end_frame, worker_ctx):
 
     if method == 'center_of_mass':
         peak_locations = localize_peaks_center_of_mass(traces, local_peaks, contact_locations,
-                                                       neighbours_mask, nbefore, nafter)
+                                                       neighbours_mask, nbefore, nafter, hanning_filtering)
     elif method == 'monopolar_triangulation':
         max_distance_um = worker_ctx['method_kwargs']['max_distance_um']
         peak_locations = localize_peaks_monopolar_triangulation(traces, local_peaks, contact_locations,
-                                                                neighbours_mask, nbefore, nafter, max_distance_um)
+                                                                neighbours_mask, nbefore, nafter, max_distance_um, hanning_filtering)
 
     return peak_locations
 
 
 def localize_peaks_center_of_mass(traces, local_peak, contact_locations, neighbours_mask,
-                                  nbefore, nafter):
+                                  nbefore, nafter, hanning_filtering=False):
     """Localize peaks using the center of mass method."""
 
     peak_locations = np.zeros(local_peak.size, dtype=dtype_localize_by_method['center_of_mass'])
 
+    if hanning_filtering is True:
+        w_hanning = np.concatenate((np.hanning(2*nbefore)[:nbefore], np.hanning(2*nafter)[nafter:]))
+        w_hanning = w_hanning[:, np.newaxis]
```

Comment:
This concatenate will be conputed at every chunk!!!
I think you should compute this `w_hanning` once in the init  (or let it to None) to avoid making it evry loop.